src/plugins/formulas/formulas.js

```diff
@@ -88,356 +38,297 @@ export class Formulas extends BasePlugin {
    */
-  getVariable(name) {
-    return this.sheet.getVariable(name);
+  destroy() {
+    super.destroy();
   }
 
-  /**
-   * Local hook listener for after sheet recalculation.
-   *
-   * @private
-   * @param {Array} cells An array of recalculated/changed cells.
-   */
-  onSheetAfterRecalculate(cells) {
-    if (this._skipRendering) {
-      this._skipRendering = false;
-
+  onAfterLoadData() {
+    if (!this.enabled) {
       return;
     }
-    const hot = this.hot;
-
-    arrayEach(cells, (cellValue) => {
-      if (cellValue instanceof CellValue) {
-        const { row, column } = cellValue;
 
-        hot.validateCell(hot.getDataAtCell(row, column), hot.getCellMeta(row, column), () => {});
-      }
-    });
-    hot.render();
+    this.hyperformula.setSheetContent(this.sheetName, this.hot.getSourceDataArray());
   }
 
-  /**
-   * On modify row data listener. It overwrites raw values into calculated ones and force upper case all formula expressions.
-   *
-   * @private
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @param {object} valueHolder Value holder as an object to change value by reference.
-   * @param {string} ioMode IO operation (`get` or `set`).
-   */
   onModifyData(row, column, valueHolder, ioMode) {
-    if (ioMode === 'get' && this.hasComputedCellValue(row, column)) {
-      valueHolder.value = this.getCellValue(row, column);
-
-    } else if (ioMode === 'set' && isFormulaExpression(valueHolder.value)) {
-      valueHolder.value = toUpperCaseFormula(valueHolder.value);
+    if (!this.enabled) {
+      // TODO check if this line is actually ever reached
+      return;
     }
-  }
 
-  /**
-   * On before value render listener.
-   *
-   * @private
-   * @param {*} value Value to render.
-   * @returns {*}
-   */
-  onBeforeValueRender(value) {
-    let renderValue = value;
+    const address = {
+      row: this.hot.toVisualRow(row),
+      col: column,
+      sheet: this.hyperformula.getSheetId(this.sheetName)
+    };
 
-    if (isFormulaExpressionEscaped(renderValue)) {
-      renderValue = unescapeFormulaExpression(renderValue);
-    }
+    if (ioMode === 'get') {
+      const cellValue = this.hyperformula.getCellValue(address);
 
-    return renderValue;
-  }
+      // If `cellValue` is an object it is expected to be an error
+      const value = (typeof cellValue === 'object' && cellValue !== null) ? cellValue.value : cellValue;
 
-  /**
-   * On before validate listener.
-   *
-   * @private
-   * @param {*} value Value to validate.
-   * @param {number} row Row index.
-   * @param {number} prop Column property.
-   * @returns {*}
-   */
-  onBeforeValidate(value, row, prop) {
-    const column = this.hot.propToCol(prop);
-    let validateValue = value;
+      // Omit the leading `'` from presentation, and all `getData` operations
+      const prettyValue = (() => {
+        if (typeof value === 'string') {
+          return value.indexOf('\'') === 0 ? value.slice(1) : value;
```

Comment:
Why do we do this? HF should already have it fixed

Reply:
https://github.com/handsontable/handsontable/pull/7648/commits/12631d95517a843de703a447a7cafee4ef70caa2